### PR TITLE
feat(ims/ecs_whole_image): support `ecs_whole_image` resource

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:

Support `ecs_whole_image` resource

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

Support `ecs_whole_image` resource

```

## PR Checklist

* [x] Tests added/passed.

```
$ export HW_ENTERPRISE_PROJECT_ID_TEST=xxxxxxxx
$ export HW_IMS_VAULT_ID=xxxxxxxxx

$ make testacc TEST="./huaweicloud/services/acceptance/ims" TESTARGS="-run TestAccEcsWholeImage_"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims -v -run TestAccEcsWholeImage_ -timeout 360m -parallel 4
=== RUN   TestAccEcsWholeImage_basic
=== PAUSE TestAccEcsWholeImage_basic
=== RUN   TestAccEcsWholeImage_withDeleteBackup
=== PAUSE TestAccEcsWholeImage_withDeleteBackup
=== CONT  TestAccEcsWholeImage_basic
=== CONT  TestAccEcsWholeImage_withDeleteBackup
--- PASS: TestAccEcsWholeImage_withDeleteBackup (418.07s)
--- PASS: TestAccEcsWholeImage_basic (702.19s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       702.290s

```

* [x] Documentation updated.
* [x] Schema updated.
* [x] CheckDeleted.

  - **a. During query operation (Read Context)**
    aa. Resource not found
image

  - **b. During delete/disassociate/unbind operation (Delete Context)**
    ba. Resource not found
image

### CODE REVIEW

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if len(imageList) < 1 {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "IMS whole image")
+	}
+
+	// For the whole image, need to use `delete_backup` to control whether to delete backup when deleting image.
+	imsClient, err := cfg.NewServiceClient(product, region)
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go]: Client is defined repeatedly.
Just use the `client` directly.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+		UpdateContext: resourceEcsWholeImageUpdate,
+		DeleteContext: resourceWholeImageDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go]: The delete timeout value is 10 minutes in document, please keep the values ​​consistent for both.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+		d.Set("os_version", image.OsVersion),
+		d.Set("active_at", image.ActiveAt),
+		d.Set("created_at", image.CreatedAt.Format(time.RFC3339)),
+		d.Set("updated_at", image.UpdatedAt.Format(time.RFC3339)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func getWholeImageInstanceIdAndVaultId(cfg *config.Config, region, backupId string) ([]string, error) {
```
Comment (deer-hang) [inline huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go]: In order to improve the utilization rate of the method, it is recommended to use the following writing method:
```suggestion
func getBackupDetail(cfg *config.Config, region, backupId string) (*backups.BackupResp, error) {
```

**Comment by jinyangyang222** [comment]
make testacc TEST="./huaweicloud/services/acceptance/ims" TESTARGS="-run TestAccEcsWholeImage_"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/ims -v -run TestAccEcsWholeImage_ -timeout 360m -parallel 4
=== RUN   TestAccEcsWholeImage_basic
=== PAUSE TestAccEcsWholeImage_basic
=== RUN   TestAccEcsWholeImage_withDeleteBackup
=== PAUSE TestAccEcsWholeImage_withDeleteBackup
=== CONT  TestAccEcsWholeImage_basic
=== CONT  TestAccEcsWholeImage_withDeleteBackup
--- PASS: TestAccEcsWholeImage_withDeleteBackup (437.69s)
--- PASS: TestAccEcsWholeImage_basic (793.21s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/ims       793.255s

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go (hunk 1 above)
Comment (jinyangyang222) [reply huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go]: fixed.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go (hunk 2 above)
Comment (jinyangyang222) [reply huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go]: fixed.

File: huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go (hunk 3 above)
Comment (jinyangyang222) [reply huaweicloud/services/ims/resource_huaweicloud_ims_ecs_whole_image.go]: fixed.

**Pull request merged**